src/server/db_slice.cc

```diff
@@ -249,6 +249,20 @@ unsigned PrimeEvictionPolicy::Evict(const PrimeTable::HotspotBuckets& eb, PrimeT
     }
 
     DbTable* table = db_slice_->GetDBTable(cntx_.db_index);
+    auto& lt = table->trans_locks;
+    string tmp;
+    string_view key = last_slot_it->first.GetSlice(&tmp);
+    // do not evict locked keys
+    if (lt.find(KeyLockArgs::GetLockKey(key)) != lt.end())
+      return 0;
+
+    // log the evicted keys to journal.
+    if (auto journal = db_slice_->shard_owner()->journal(); journal) {
+      ArgSlice delete_args{string(key)};
```

Comment:
nit: ArgSlice is Span<string_view> so it can hold just `key`

Reply:
fixed.